decoder/src/lib.rs

```diff
@@ -119,7 +119,7 @@ struct BitflagsKey {
 }
 
 #[derive(Copy, Clone, Debug, PartialEq)]
-enum Encoding {
+pub enum Encoding {
```

Comment:
although all of `defmt-decoder` is unstable making this enum public is a semver hazard because now adding a variant is a breaking change in the API and that requires a semver bump. could you add a `#[non_exhaustive]` attribute here to prevent that potential issue?

(we could even go further and prevent matching on the enum by hiding it inside a newtype `pub struct Encoding(EncodingKind); enum EncodingKind { .. }` but that may be overkill)

Reply:
> could you add a `#[non_exhaustive]` attribute here to prevent that potential issue?

Done!